src/diffusers/onnx_utils.py

```diff
@@ -0,0 +1,187 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Inc. team.
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import os
+import shutil
+from pathlib import Path
+from typing import Optional, Union
+
+import numpy as np
+
+from huggingface_hub import hf_hub_download
+
+from .utils import is_onnx_available, logging
+
+
+if is_onnx_available():
+    import onnxruntime as ort
+
+
+ONNX_WEIGHTS_NAME = "model.onnx"
+
+
+logger = logging.get_logger(__name__)
+
+
+class OnnxModel:
```

Comment:
turbo nit: ONNX and ONNX Runtime are commonly confused as being the same thing (the former being a format, the latter a runtime)

it would be technically more correct to call this `ORTModel` or `OnnxRuntimeModel` since the `__call__` method is passing the ONNX model through ONNX Runtime.

(no need to follow this advice, just noting it since the discussion also came up in `optimum` :))

Reply:
Makes sense to me to call it `OnnxRuntimeModel` -> your call though @anton-l :-)

Reply:
If we are going to change the name, I prefer `OnnxRuntimeModel` too (rather than `ORTModel`, even if people use ORT all the time).

Reply:
Opted for `OnnxRuntimeModel`!